README.md

```diff
@@ -1,104 +1,347 @@
+    'longitude': 7836,
+  };
+
+  print('$workivaAmes => ${schema.validate(workivaAmes)}'); // true
+  print('$nowhereville => ${schema.validate(nowhereville)}'); // false
+
+  // Exit the process cleanly (VM Only).
+  exit(0);
+}
+```
+
+### Asynchronous Creation, with custom remote $refs:
+
+If you have nested $refs that are either non-HTTP URIs or non-publically-accessible HTTP $refs, you can supply an `RefProviderAsync` to `createSchemaAsync`, and perform any custom logic you need.
+
+#### Example
+
+```dart
+import 'dart:io';
+import 'dart:async';
+import 'package:dart2_constant/convert.dart';
+
+import 'package:json_schema/json_schema.dart';
+
+// For VM:
+import 'package:json_schema/vm.dart';
+
+// For Browser:
+// import 'package:json_schema/browser.dart';
+
+main() async {
+  // For VM:
+  configureJsonSchemaForVm();
+
+  // For Browser:
+  // configureJsonSchemaForBrowser();
+
+  final referencedSchema = {
+    r"$id": "https://example.com/geographical-location.schema.json",
+    r"$schema": "http://json-schema.org/draft-06/schema#",
+    "title": "Longitude and Latitude",
+    "description": "A geographical coordinate on a planet (most commonly Earth).",
+    "required": ["latitude", "longitude"],
+    "type": "object",
+    "properties": {
+      "name": {"type": "string"},
+      "latitude": {"type": "number", "minimum": -90, "maximum": 90},
+      "longitude": {"type": "number", "minimum": -180, "maximum": 180}
+    }
+  };
+
+  final RefProviderAsync refProvider = (String ref) async {
+    final Map references = {
+      'https://example.com/geographical-location.schema.json': JsonSchema.createSchema(referencedSchema),
+    };
+
+    if (references.containsKey(ref)) {
+      // Silly example that adds a 1 second delay.
+      // In practice, you could make any service call here,
+      // parse the results into a schema, and return.
+      await new Future.delayed(new Duration(seconds: 1));
+      return references[ref];
+    }
+
+    // Fall back to default URL $ref behavior
+    return await JsonSchema.createSchemaFromUrl(ref);
+  };
+
+  final schema = await JsonSchema.createSchemaAsync({
+    'type': 'array',
+    'items': {r'$ref': 'https://example.com/geographical-location.schema.json'}
+  }, refProvider: refProvider);
+
+  final workivaLocations = [
+    {
+      'name': 'Ames',
+      'latitude': 41.9956731,
+      'longitude': -93.6403663,
+    },
+    {
+      'name': 'Scottsdale',
+      'latitude': 33.4634707,
+      'longitude': -111.9266617,
+    }
+  ];
+
+  final badLocations = [
+    {
+      'name': 'Bad Badlands',
+      'latitude': 181,
+      'longitude': 92,
+    },
+    {
+      'name': 'Nowhereville',
+      'latitude': -2000,
+      'longitude': 7836,
+    }
+  ];
+
+  print('${json.encode(workivaLocations)} => ${schema.validate(workivaLocations)}');
+  print('${json.encode(badLocations)} => ${schema.validate(badLocations)}');
+
+  exit(0);
+}
+```
+
+## How To Use Schema Information
```

Comment:
Fantastic job on documentation in this file!  💯

Reply:
Thanks!